tests/sources/test_dropbox_aiohttp.py

```diff
@@ -0,0 +1,809 @@
+        )
+        await source.ping()
+
+
+@pytest.mark.asyncio
+@patch("connectors.sources.dropbox_aiohttp.RETRY_INTERVAL", 0)
+async def test_api_call_negative():
+    source = create_source(DropboxDataSource)
+    source.dropbox_client.retry_count = 4
+    source.dropbox_client._set_access_token = AsyncMock()
+
+    with patch.object(
+        aiohttp.ClientSession, "post", side_effect=Exception("Something went wrong")
+    ):
+        with pytest.raises(Exception):
+            await anext(
+                source.dropbox_client.api_call(
+                    base_url=HOST_URLS["FILES_FOLDERS_HOST_URL"],
+                    url_name=PING,
+                    data=json.dumps(None),
+                )
+            )
+
+    with patch.object(
+        aiohttp.ClientSession, "post", side_effect=ServerDisconnectedError()
+    ):
+        with pytest.raises(Exception):
+            await anext(
+                source.dropbox_client.api_call(
+                    base_url=HOST_URLS["FILES_FOLDERS_HOST_URL"],
+                    url_name=PING,
+                    data=json.dumps(None),
+                )
+            )
+
+
+@pytest.mark.asyncio
+async def test_api_call():
+    source = create_source(DropboxDataSource)
+    source.dropbox_client._set_access_token = AsyncMock()
+
+    with patch.object(
+        aiohttp.ClientSession,
+        "post",
+        return_value=get_json_mock(MOCK_CURRENT_USER, 200),
+    ):
+        EXPECTED_RESPONSE = {
+            "account_id": "acc_id:1234",
+            "name": {
+                "given_name": "John",
+                "surname": "Wilber",
+                "display_name": "John Wilber",
+                "abbreviated_name": "JW",
+            },
+            "email": "john.wilber@abcd.com",
+            "country": "US",
+        }
+        response = await anext(
+            source.dropbox_client.api_call(
+                base_url=HOST_URLS["FILES_FOLDERS_HOST_URL"],
+                url_name=PING,
+                data=json.dumps(None),
+            )
+        )
+        actual_response = await response.json()
+        assert actual_response == EXPECTED_RESPONSE
+
+
+@pytest.mark.asyncio
+async def test_set_access_token_when_token_expires_at_is_str():
+    source = create_source(DropboxDataSource)
+    source.dropbox_client.token_expiration_time = "2023-02-10T09:02:23.629821"
+    mock_token = {"access_token": "test2344", "expires_in": "1234555"}
+    async_response_token = get_json_mock(mock_token, 200)
+
+    with patch.object(aiohttp.ClientSession, "post", return_value=async_response_token):
+        actual_response = await source.dropbox_client._set_access_token()
+        assert actual_response is None
+
+
+@pytest.fixture
+def patch_default_wait_multiplier():
+    with mock.patch("connectors.sources.dropbox.RETRY_INTERVAL", 0):
+        yield
+
+
+@pytest.mark.asyncio
+@mock.patch("connectors.utils.apply_retry_strategy")
+async def test_api_call_when_token_is_expired(
```

Comment:
Shouldn't it try to refresh? Isn't there a a configured refresh token?

Reply:
I see! let me give it a try